lib/Query.js

```diff
@@ -83,7 +83,11 @@ Query.prototype.exec = function (next) {
   };
 
   var indexName, index;
-  if(schema.hashKey.name !== this.query.hashKey.name) {
+  // Check both hash key and range key in the query to see if they do not match
+  // the hash and range key on the primary table.  If they don't match then we
+  // can look for a secondary index to query.
+  if(schema.hashKey.name !== this.query.hashKey.name ||
```

Comment:
Just thought of an issue -- we need to add a check to skip this if `.IndexName` is already defined. Else it will overwrite what was configured with `.using`.